start.sh

```diff
@@ -112,39 +237,135 @@ IPFS_UA_GATEWAY_URL="${DEFAULT_IPFS_UA_GATEWAY_URL}"
 EOI
     fi
 
-fi
+    export_save_variable PROFILES "${PROFILES}"
+    export_save_variable COMPOSE_FILES "${COMPOSE_FILES}"
+}
 
-set -a; source .env-saved; set +a
+###################################################################################
+# local_setup_tasks
+#
+# Description: Perform tasks after starting services, such as setting up environments
+#
+###################################################################################
+function local_setup_tasks() {
+    if [ "${SKIP_CHAIN_SETUP}" != true ] && [ "${TESTNET_ENV}" != true ]; then
+        # Wait 1 minute for Frequency node to be ready
+        health_attempts=0
+        while (( health_attempts < 60 )) && ! is_frequency_ready; do
+            (( health_attempts += 1 ))
+            echo "Waiting for Frequency node to respond..."
+            sleep 1
+        done
 
-if [[ ! $TESTNET_ENV =~ ^[Yy]$ ]]; then
-    # Start specific services in detached mode
-    echo -e "\nStarting local frequency services..."
-    docker compose up -d frequency
-fi
+        if is_frequency_ready; then
+            # Run npm run local:init
+            echo "Running local setup script to provision Provider with capacity, etc..."
+            npm run setup:content-publishing:chain
```

Comment:
Is this the right setup script for all cases?
I guess we probably need to consolidate these...

Reply:
It's the same code used in social-app-template and I didn't want to add a new file.
Yeah, we could audit those setups and see which varies from a common base.